src/components/GlobalHeader/RightContent.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import { FormattedMessage, formatMessage } from 'umi/locale'
+import { Spin, Tag, Menu, Icon, Avatar, Tooltip } from 'antd'
+import moment from 'moment'
+import groupBy from 'lodash/groupBy'
+import { NoticeIcon } from 'ant-design-pro'
+import HeaderSearch from '../HeaderSearch'
+import HeaderDropdown from '../HeaderDropdown'
+import SelectLang from '../SelectLang'
+import styles from './index.less'
+
+export default function GlobalHeaderRight (props) {
+  const {
+    currentUser,
+    fetchingNotices,
+    onNoticeVisibleChange,
+    onMenuClick,
+    onNoticeClear,
+    theme,
+    notices = [],
+    dispatch
+  } = props
+
+  function getNoticeData () {
+    if (notices.length === 0) {
+      return {}
+    }
+    const newNotices = notices.map(notice => {
+      const newNotice = { ...notice }
+      if (newNotice.datetime) {
+        newNotice.datetime = moment(notice.datetime).fromNow()
+      }
+      if (newNotice.id) {
+        newNotice.key = newNotice.id
+      }
+      if (newNotice.extra && newNotice.status) {
+        const color = {
+          todo: '',
+          processing: 'blue',
+          urgent: 'red',
+          doing: 'gold'
+        }[newNotice.status]
+        newNotice.extra = (
+          <Tag color={color} style={{ marginRight: 0 }}>
+            {newNotice.extra}
+          </Tag>
+        )
+      }
+      return newNotice
+    })
+    return groupBy(newNotices, 'type')
+  }
+
+  function getUnreadData (noticeData) {
+    const unreadMsg = {}
+    Object.entries(noticeData).forEach(([key, value]) => {
+      if (!unreadMsg[key]) {
+        unreadMsg[key] = 0
+      }
+      if (Array.isArray(value)) {
+        unreadMsg[key] = value.filter(item => !item.read).length
+      }
+    })
+    return unreadMsg
+  }
```

Comment:
As above, let's move it to the module scope.
I don't like that a closure function modifies outside data (the `forEach`). Hence I suggest for loop. In functional world this things are really alarming.
I know that this was already in the original code, so if you want we can create a new task and solve it in a new task.